packages/appkit/src/universal-adapter/client.ts

```diff
@@ -464,28 +462,24 @@ export class UniversalAdapterClient {
     const chainChanged = (chainId: number | string) => {
       // eslint-disable-next-line eqeqeq
       const caipNetwork = this.caipNetworks.find(c => c.chainId == chainId)
-      const isSameNetwork =
-        caipNetwork?.chainId === ChainController.state.activeCaipNetwork?.chainId
+      const currentCaipNetwork = this.appKit?.getCaipNetwork()
+
+      if (!caipNetwork) {
+        NetworkController.setActiveCaipNetwork({
+          chainId: Number(chainId),
+          id: `eip155:${chainId}`,
```

Comment:
Why we setting only `eip155`? We should refactor this as that it'll get different namespaces as well. 

What could be the `chainId` other than EVM or Solana chain IDs?